sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -73,36 +108,32 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
             }
 
             string version = versionAttribute.InformationalVersion;
-
             string assemblyName = clientAssembly.GetName().Name!;
-            if (assemblyName.StartsWith(PackagePrefix, StringComparison.Ordinal))
-            {
-                assemblyName = assemblyName.Substring(PackagePrefix.Length);
-            }
 
-            int hashSeparator = version.IndexOfOrdinal('+');
+            // Azure SDK specific: Convert assembly name to azsdk-net- format
+            assemblyName = TransformToAzureSdkAssemblyName(assemblyName);
+
+            int hashSeparator = version.IndexOf('+');
             if (hashSeparator != -1)
             {
                 version = version.Substring(0, hashSeparator);
             }
-            runtimeInformation ??= new RuntimeInformationWrapper();
 
-            // RFC 9110 section 5.5 https://www.rfc-editor.org/rfc/rfc9110.txt#section-5.5 does not require any specific encoding : "Fields needing a greater range of characters
-            // can use an encoding, such as the one defined in RFC8187." RFC8187 is targeted at parameter values, almost always filename, so using url encoding here instead, which is
-            // more widely used. Since user-agent does not usually contain non-ascii, only encode when necessary.
-            // This was added to support operating systems with non-ascii characters in their release names.
-            string osDescription;
+            string osDescription = runtimeInformation.OSDescription;
+            string frameworkDescription = runtimeInformation.FrameworkDescription;
+
+            // URL encode non-ASCII characters in OS description
 #if NET8_0_OR_GREATER
-            osDescription = Ascii.IsValid(runtimeInformation.OSDescription) ? runtimeInformation.OSDescription : WebUtility.UrlEncode(runtimeInformation.OSDescription);
+            osDescription = Ascii.IsValid(osDescription) ? osDescription : WebUtility.UrlEncode(osDescription);
 #else
-            osDescription = ContainsNonAscii(runtimeInformation.OSDescription) ? WebUtility.UrlEncode(runtimeInformation.OSDescription) : runtimeInformation.OSDescription;
+            osDescription = ContainsNonAscii(osDescription) ? WebUtility.UrlEncode(osDescription) : osDescription;
 #endif
 
-            var platformInformation = EscapeProductInformation($"({runtimeInformation.FrameworkDescription}; {osDescription})");
+            var platformInformation = EscapeProductInformation($"({frameworkDescription}; {osDescription})");
 
             return applicationId != null
-                ? $"{applicationId} azsdk-net-{assemblyName}/{version} {platformInformation}"
-                : $"azsdk-net-{assemblyName}/{version} {platformInformation}";
+                ? $"{applicationId} {assemblyName}/{version} {platformInformation}"
+                : $"{assemblyName}/{version} {platformInformation}";
```

Comment:
We should delete all of the now unused code. We should also add additional unit tests.

Reply:
We should not duplicate System.ClientModel's private methods for testing purposes. We should instead use Moq or some other test-only method for producing the test data.

Reply:
Please remove GenerateUserAgentStringWithCustomRuntimeInfo and the RuntimeInformationWrapper from the Azure.Core usage. Instead, move the RuntimeInformationWrapper into SCM and include an internal method that takes runtimeInformation as a parameter. Move all tests that leveraged the MockRuntimeInformation into the UserAgentPolicyTests in SCM.